app.rb

```diff
@@ -14,5 +14,18 @@ class App < Sinatra::Base
     payload = JSON.parse request.body.read
 
     Net::HTTP.get(URI(payload['SubscribeURL'])) if payload['Type'] == 'SubscriptionConfirmation'
+    signup_user(JSON.parse(payload['Message'])) if payload['Type'] == 'Notification'
+    ""
+  end
+
+private
+
+  def signup_user(ses_notification)
+    from_address = ses_notification['commonHeaders']['from'][0]
+    User.new.generate(email: from_address) if authorised_email_domains_regex.match? from_address
```

Comment:
not a fan of this as a guard clause, I'd suggest `if authorised_email_domain?(from_address)` and do the regex match in there

Reply:
Agreed.